girder/api/v1/file.py

```diff
@@ -232,24 +228,11 @@ def readChunk(self, upload, offset, params):
         the writer of the chunk is the same as the person who initiated the
         upload. The passed offset is a verification mechanism for ensuring the
         server and client agree on the number of bytes sent/received.
-
-        This method accepts both the legacy multipart content encoding, as
-        well as passing offset and uploadId as query parameters and passing
-        the chunk as the body, which is the recommended method.
-
-        .. deprecated :: 2.2.0
         """
         if 'chunk' in params:
```

Comment:
When would this case still occur?

Reply:
It happens in the direct to S3 upload behavior. Which is not an ideal design, but that's what it's for.

Reply:
Can you add a code comment to explain this?
* If we see the undocumented `chunk` query string parameter, then we abort trying to read the body, use the query string value as `chunk`, and pass it through to `Upload().handleChunk`
* This case is typically used by the direct S3 upload process..